From f6d88d293a238fa962e6d4040fbf4e2ead4ddf3d Mon Sep 17 00:00:00 2001
From: Travis Ellett <1130469+tellett@users.noreply.github.com>
Date: Tue, 27 Oct 2026 11:53:17 -0600
Subject: [PATCH 1/2] Restore --experimental_remote_analysis_cache

Reverts db6088a6e07, which moved --experimental_remote_analysis_cache and
--experimental_remote_analysis_unreachable_cache_retry_interval into
BazelRulesModule's graveyard as no-ops. Both are declared again in
RemoteAnalysisCachingOptions, as abstract getters to match the rest of
that class, and removed from the graveyard so the names do not collide.

This reverts commit db6088a6e0712de3d60b6ca5be3e2062b3ca257c.
---
 .../lib/bazel/rules/BazelRulesModule.java     | 20 ----------------
 .../RemoteAnalysisCachingOptions.java         | 23 +++++++++++++++++++
 2 files changed, 23 insertions(+), 20 deletions(-)

diff --git a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRulesModule.java b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRulesModule.java
index f35b0c304f7..6ba24c80467 100644
--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRulesModule.java
+++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRulesModule.java
@@ -589,26 +589,6 @@ public final class BazelRulesModule extends BlazeModule {
         help = "No-op. Will be removed soon.")
     public abstract boolean getSkipTtvs();
 
-    @Deprecated
-    @Option(
-        name = "experimental_remote_analysis_cache",
-        defaultValue = "",
-        documentationCategory = OptionDocumentationCategory.UNDOCUMENTED,
-        effectTags = {OptionEffectTag.NO_OP},
-        metadataTags = {OptionMetadataTag.DEPRECATED},
-        help = "No-op.")
-    public abstract String getRemoteAnalysisCache();
-
-    @Deprecated
-    @Option(
-        name = "experimental_remote_analysis_unreachable_cache_retry_interval",
-        defaultValue = "",
-        documentationCategory = OptionDocumentationCategory.UNDOCUMENTED,
-        effectTags = {OptionEffectTag.NO_OP},
-        metadataTags = {OptionMetadataTag.DEPRECATED},
-        help = "No-op")
-    public abstract String getUnreachableCacheRetryInterval();
-
     @Deprecated
     @Option(
         name = "remote_analysis_json_log",
diff --git a/src/main/java/com/google/devtools/build/lib/skyframe/serialization/analysis/RemoteAnalysisCachingOptions.java b/src/main/java/com/google/devtools/build/lib/skyframe/serialization/analysis/RemoteAnalysisCachingOptions.java
index efff2c54f73..c2a8d3882dc 100644
--- a/src/main/java/com/google/devtools/build/lib/skyframe/serialization/analysis/RemoteAnalysisCachingOptions.java
+++ b/src/main/java/com/google/devtools/build/lib/skyframe/serialization/analysis/RemoteAnalysisCachingOptions.java
@@ -19,12 +19,14 @@ import com.google.common.base.Strings;
 import com.google.common.hash.HashCode;
 import com.google.common.hash.Hashing;
 import com.google.devtools.common.options.Converter;
+import com.google.devtools.common.options.Converters.DurationConverter;
 import com.google.devtools.common.options.Option;
 import com.google.devtools.common.options.OptionDocumentationCategory;
 import com.google.devtools.common.options.OptionEffectTag;
 import com.google.devtools.common.options.OptionsBase;
 import com.google.devtools.common.options.OptionsClass;
 import com.google.devtools.common.options.OptionsParsingException;
+import java.time.Duration;
 import javax.annotation.Nullable;
 
 /** Options for caching analysis results remotely. */
@@ -104,6 +106,27 @@ public abstract class RemoteAnalysisCachingOptions extends OptionsBase {
     }
   }
 
+  @Option(
+      name = "experimental_remote_analysis_cache",
+      defaultValue = "",
+      documentationCategory = OptionDocumentationCategory.UNDOCUMENTED,
+      effectTags = {OptionEffectTag.BAZEL_INTERNAL_CONFIGURATION},
+      help = "The URL for the remote analysis caching backend.")
+  public abstract String getRemoteAnalysisCache();
+
+  // TODO: b/443947033 - add a way to disable retries
+  @Option(
+      name = "experimental_remote_analysis_unreachable_cache_retry_interval",
+      documentationCategory = OptionDocumentationCategory.UNDOCUMENTED,
+      effectTags = {OptionEffectTag.BAZEL_INTERNAL_CONFIGURATION},
+      defaultValue = "250ms",
+      converter = DurationConverter.class,
+      help =
+          "How long to wait before retrying a cache get request that failed due to an UNREACHABLE"
+              + " channel. This is a workaround for the client library reporting 'ready' "
+              + "prematurely.")
+  public abstract Duration getUnreachableCacheRetryInterval();
+
   @Option(
       name = "experimental_analysis_cache_key_distinguisher_for_testing",
       defaultValue = "null",

base-commit: e6042f9c2260b2526ac3fb75fc91ed81538eeecb
-- 
2.55.0

